lib/boom_notifier/error_info.ex

```diff
@@ -2,56 +2,55 @@ defmodule ErrorInfo do
   @moduledoc false
 
   # The ErrorInfo struct holds all the information about the exception.
-  # It includes the error message, the stacktrace and context information
+  # It includes the error message, the stacktrace, context information
   # (information about the request, the current controller and action,
-  # among other things).
+  # among other things) and custom data depending on the configuration.
 
   @enforce_keys [:reason, :stack, :timestamp]
-  defstruct [:name, :reason, :stack, :controller, :action, :request, :timestamp]
+  defstruct [:name, :reason, :stack, :controller, :action, :request, :timestamp, :data]
+
+  @type option ::
+          :logger
+          | [logger: [fields: list(atom())]]
+          | :assigns
+          | [assigns: [fields: list(atom())]]
+  @type custom_data_strategy_type :: :nothing | option | [option]
 
   @spec build(
           %{
             required(:reason) => any(),
             required(:stack) => Exception.stacktrace(),
             optional(any()) => any()
           },
-          map()
+          map(),
+          custom_data_strategy_type
         ) :: {atom(), %ErrorInfo{}}
-  def build(%{reason: reason, stack: stack} = error, conn) do
-    {get_reason(error), build_info(reason, stack, conn)}
-  end
-
-  defp build_info(%name{message: reason}, stack, conn) do
-    %{build_without_name(reason, stack, conn) | name: name}
-  end
-
-  defp build_info(%{message: reason}, stack, conn) do
-    %{build_without_name(reason, stack, conn) | name: "Error"}
-  end
-
-  defp build_info(reason, stack, conn) when is_binary(reason) do
-    build_info(%{message: reason}, stack, conn)
-  end
-
-  defp build_info(reason, stack, conn) do
-    build_info(%{message: inspect(reason)}, stack, conn)
-  end
-
-  defp get_reason(%{reason: %name{}}), do: name
-  defp get_reason(%{error: %{kind: kind}}), do: kind
-  defp get_reason(_), do: :error
+  def build(%{reason: reason, stack: stack} = error, conn, custom_data_strategy) do
+    {error_reason, error_name} = error_reason(reason)
 
-  defp build_without_name(reason, stack, conn) do
-    %ErrorInfo{
-      reason: reason,
+    error_info = %ErrorInfo{
+      reason: error_reason,
       stack: stack,
       controller: get_in(conn.private, [:phoenix_controller]),
       action: get_in(conn.private, [:phoenix_action]),
       request: build_request_info(conn),
-      timestamp: DateTime.utc_now()
+      timestamp: DateTime.utc_now(),
+      name: error_name,
+      data: build_custom_data(conn, custom_data_strategy)
```

Comment:
What do you folks think about havings this field rename to `custom_data` or `metadata`?

Reply:
I like this, data can be too generic. `metadata` sounds better to me but probably we should rename the functions that build it too and all the passed variables

Reply:
I like `metadata`. Thanks!